patches/minecraft/net/minecraft/inventory/Slot.java.patch

```diff
@@ -9,7 +9,18 @@
      }
  
      public ItemStack func_75209_a(int p_75209_1_)
-@@ -113,4 +113,74 @@
+@@ -105,7 +105,9 @@
+ 
+     public boolean func_82869_a(EntityPlayer p_82869_1_)
+     {
+-        return true;
++        ItemStack itemStack = this.func_75211_c();
++        net.minecraft.util.EnumActionResult cancelResult = net.minecraftforge.common.ForgeHooks.onItemTake(p_82869_1_, itemStack);
```

Comment:
As I can see, this method is frequently overridden by mods without calling super. You might consider another place to patch so that this event remains useful.

Reply:
Is that so... That's a shame because after looking through many classes this seemed to be the easiest implementation without modifying too many files. Based on what you said about overriding, do you think it would be acceptable if I add a new method that uses the event, and calls `canTakeStack`? I would have to go and replace all the occurrences of `canTakeStack` in Minecraft's code though, but I don't think there are too many.

Reply:
Alternatively, I could just add the event code at each occurrence of `canTakeStack`, but I don't think that's the right way to go.

Reply:
I don't think you should do that either.

(a) Adding a new method means you're incompatible with all inventories which call canTakeStack on their own.
(b) Replacing each occurence won't cover mods.

I think that you should do a clean implementation and simply wait for 1.13 with it. There's no way to do this cleanly in 1.12 this late into the game.

Reply:
Hmm... That's a shame. For now do you think I should temporarily do something in my own mod that has some work around for this?